src/interpreter/type_cast.rs

```diff
@@ -0,0 +1,170 @@
+use crate::interpreter::types::{Numeric, PactType, StringLike};
+use std::any::Any;
+use std::convert::TryFrom;
+
+#[derive(Debug, PartialEq)]
+pub enum PactConversionErr {
+    NegativeInteger,
+    Overflow,
+    UnknownType,
+}
+
+/// A catch-all conversion trait which tries to turn any given `value` into the implementing type
+pub trait AnyTryInto<'a>: Sized {
+    fn any_try_into(value: &'a dyn Any) -> Result<Self, PactConversionErr>;
+}
+
+/// AnyTryInto implementation for PactType
+impl<'a> AnyTryInto<'a> for PactType<'a> {
+    fn any_try_into(value: &'a dyn Any) -> Result<PactType<'a>, PactConversionErr> {
+        // TODO: refactor the below repetiion using macros
+        // Signed integer type casting into PactType
+        if value.is::<i8>()
```

Comment:
Is there a reason to do an `is` check before down casting?

Reply:
Nope there is no need. All the `is` conditionals are removed